nomad/node_endpoint.go

```diff
@@ -1143,7 +1143,14 @@ func (n *Node) DeriveVaultToken(args *structs.DeriveVaultTokenRequest,
 		if e == nil {
 			return
 		}
-		reply.Error = structs.NewRecoverableError(e, recoverable).(*structs.RecoverableError)
+		re, ok := e.(structs.Recoverable)
```

Comment:
Assert to `*structs.RecoverableError` here and skip the second assertion below. One less assertion and there are multiple types that implement `structs.Recoverable` (even if right now only one of them is used here).